examples/test-contract/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::{env, near_bindgen};
+
+#[global_allocator]
+static ALLOC: near_sdk::wee_alloc::WeeAlloc<'_> = near_sdk::wee_alloc::WeeAlloc::INIT;
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct TestContract {}
+
+impl Default for TestContract {
+    fn default() -> Self {
+        Self {}
+    }
+}
+
+#[near_bindgen]
+impl TestContract {
+    #[init]
+    pub fn new() -> Self {
+        assert!(!env::state_exists(), "Already initialized");
+        Self {}
+    }
+
+    #[init]
```

Comment:
Why the second init?

Reply:
The idea for `migrate` with `init` is to be able to upgrade contract and migrate it from the old data scheme to the new data scheme. Assuming we can't do this natively with borsh, we can use `migrate` call to manually deserialize the state using the old struct/schema and then save the state serialized with the current struct/schema.

While `new` is for original initialization, when the state is not available.